source/helpers/function.ts

```diff
@@ -0,0 +1,107 @@
+import * as cheerio from 'cheerio';
+
+import { codeBlock } from './components';
+
+const links: Record<string, string> = {
+  number: '/documentation/values/numbers',
+  string: '/documentation/values/strings',
+  'quoted string': '/documentation/values/strings#quoted',
+  'unquoted string': '/documentation/values/strings#unquoted',
+  color: '/documentation/values/colors',
+  list: '/documentation/values/lists',
+  map: '/documentation/values/maps',
+  boolean: '/documentation/values/booleans',
+  null: '/documentation/values/null',
+  function: '/documentation/values/functions',
+  selector: '/documentation/modules/selector#selector-values',
+};
+
+const returnTypeLink = (returnType: string) =>
+  returnType
+    .split('|')
+    .map((type) => {
+      type = type.trim();
+      const link = links[type];
+      if (!link) {
+        throw new Error(`Unknown type ${type}`);
+      }
+      return `<a href="${link}">${type}</a>`;
+    })
+    .join(' | ');
+
+/** Renders API docs for a Sass function (or mixin).
+ *
+ * The function's name is parsed from the signature. The API description is
+ * passed as a Markdown block. If `returns:type` is passed as the last argument,
+ * it's included as the function's return type.
+ *
+ * Multiple signatures may be passed, in which case they're all included in
+ * sequence.
+ */
+export function _function(content: string, ...signatures: string[]) {
+  // Parse the last argument as the return type, if it's present
+  const returns = signatures.at(-1)?.match(/returns?:\s*(.*)/)?.[1];
+  if (returns) signatures.pop();
+
+  // Highlight each signature
+  const names: string[] = [];
+  const highlightedSignatures = signatures.map((signature) => {
+    signature = signature.replace(/!!!/g, '\n'); // Hack to allow newlines in the signature
+    const [name] = signature.split('(', 2);
+    const nameWithoutNamespace = name.split('.').at(-1) || name;
+    const html = codeBlock(`@function ${signature}`, 'scss');
```

Comment:
We're throwing away some of the functionality of `codeBlock` in this case (e.g. the wrapping `<pre><code>` els), but I guess that doesn't really hurt.